pkg/container-collection/container-collection.go

```diff
@@ -339,7 +370,12 @@ func (cc *ContainerCollection) EnrichByMntNs(event *eventtypes.CommonData, mount
 func (cc *ContainerCollection) EnrichByNetNs(event *eventtypes.CommonData, netnsid uint64) {
 	event.Node = cc.nodeName
 
-	containers := cc.LookupContainersByNetns(netnsid)
+	var containers []*Container
+
+	containers = append(containers, lookupContainersByNetns(&cc.containers, netnsid)...)
+	if cc.cachedContainers != nil {
```

Comment:
For `EnrichByMntNs`, you also test `container == nil` but not here. It seems reasonable because there could be several containers in the same netns, so we should look into `cc.cachedContainers` even if we find a container in `cc.containers` already. But that makes the semantic subtle.

`EnrichByNetNs` is only used by the network tracer (pkg/gadgets/trace/network/tracer/tracer.go) and it has its own cache for enriching events of terminated containers. That cache works differently than this one. Do you intend to keep both?

Reply:
hum, it's a good question. I didn't take into account the cache mechanism implemented in trace network. I don't think we can combine two approaches easily as the one used in this PR is to hold the container in memory long enough to allow the tracer to read and enrich an event from the ring buffer. On the other hand, in the trace network gadget that's different, we need to have a cache until the user calls `Pop()`, and it could be a much longer time span.

Reply:
Also, the current design I proposed here is wrong, it could happen that a pod is removed and then a new one is created reusing the same network namespace inode id, hence there is a timestamp where both pods will be in memory and the enrichment will give wrong results. A possibility is to also keep a reference of the net ns id to avoid it being reused, but I'm getting worried about the number of opened file descriptors we keep. 

I'd propose to remove the changes introduced to `EnrichByNetNs` in this PR and discuss that later on, specially because I'm wondering how https://github.com/inspektor-gadget/inspektor-gadget/pull/1201 will impact all the enrich logic in networking gadgets.  WDYT?